infra/event-sync/src/main/java/com/simprints/infra/eventsync/sync/up/tasks/EventUpSyncTask.kt

```diff
@@ -229,8 +237,10 @@ internal class EventUpSyncTask @Inject constructor(
      *
      * Additionally emits the number of events in each scope to be used for progress tracking.
      */
-    private suspend fun FlowCollector<Int>.getClosedScopesForType(type: EventScopeType) =
-        eventRepository.getClosedEventScopes(type).associateWith {
+    private suspend fun FlowCollector<Int>.getClosedScopesForType(
+        type: EventScopeType,
+        limit: Int,
+    ) =  eventRepository.getClosedEventScopes(type, limit).associateWith {
             try {
                 eventRepository.getEventsFromScope(it.id)
                     .also { listOfEvents -> emit(listOfEvents.size) }
```

Comment:
@BurningAXE here you will emit to the progress bar only the partial events count for the selected scope not all scopes

Reply:
Yes, that's the only drawback I could think of - it will lead to jumpy progress bar, too.

Reply:
@BurningAXE Overall, well done in such a short time.